### ISSUE TITLE
line not wrap and place space character in the right

### ISSUE THREAD

**Issue by grunmin** [xtermjs/xterm.js#1133]
## TL;DR
I found the reason.
Size option of ssh screen(backend) should be the same of xterm's, or server will send `\n` if index of current character match the width. And make sure backend return `\r\n` instead of `\n`.

--------
I had issue some problems when i use xterm.js in React env.

When I use `attach` to attach a websocket, xterm will always place (9) space character in the right of container. And if I then add `fit`, current line will not wrap instead replace old character.

![screen_cap_16](https://user-images.githubusercontent.com/4323398/33307012-f25b7f34-d44f-11e7-943a-b8d42f70494a.gif)

## Details
- Browser and browser version: Chrome 62
- OS version: Windows 10, Mac as well
- xterm.js version: 2.9.1, 2.9.2 as well

### Steps to reproduce
1. Code follow one of these two repos  in [storybook](https://github.com/storybooks/storybook). (1). [react-terminal](https://github.com/leinue/react-terminal), 
(2)[react-xterm](https://github.com/nicescale/react-xterm)
2. run xterm.js demo in linux to provide ws server
3. specify socketUrl and connect succeed
4. input something and issue that problem

**Comment by mofux** [comment]
As you already discovered, the xterm.js screen dimensions should match the screen dimensions of the baking terminal process. Can this issue be closed?
